src/main/java/gyro/google/dns/DnsPolicyNetwork.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.dns;
+
+import java.util.Optional;
+
+import com.google.api.services.dns.model.PolicyNetwork;
+import gyro.core.resource.Diffable;
+import gyro.google.Copyable;
+import gyro.google.Requestable;
+import gyro.google.compute.NetworkResource;
+
+public class DnsPolicyNetwork extends Diffable implements Copyable<PolicyNetwork>,
+    Requestable<PolicyNetwork> {
+
+    private NetworkResource network;
+
+    /**
+     * The VPC network to bind to. The value may be ``null``.
+     *
+     * @resource gyro.google.compute.NetworkResource
+     */
+    public NetworkResource getNetwork() {
+        return network;
+    }
+
+    public void setNetwork(NetworkResource network) {
+        this.network = network;
+    }
+
+    @Override
+    public void copyFrom(PolicyNetwork model) {
+        String networkUrl = model.getNetworkUrl();
+
+        if (networkUrl != null) {
+            setNetwork(findById(NetworkResource.class, networkUrl.substring(networkUrl.lastIndexOf("/") + 1)));
+        }
+    }
+
+    @Override
+    public PolicyNetwork copyTo() {
+        PolicyNetwork policyNetwork = new PolicyNetwork();
+        NetworkResource network = getNetwork();
+
+        if (network != null) {
+            policyNetwork.setNetworkUrl(network.getSelfLink());
```

Comment:
The NetworkResource has the `@Id` field set on the `name` attribute. So while configuring this complex type if network is assigned a string, which means a network which is not managed by gyro, it will create an empty NetworkResource model and set the `name` attribute to value being passed. Correspondingly the `self-link` field will be null and cause this to fail.

The `@Id` field of the NetworkResource should instead be put on the `self-link` attribute, so if an unmanaged network is assigned to the `network` attribute of this complex type, and empty `NetworkResource` object is created with the `self-link` attribute set to be used.

This change will mean that in `copyFrom` the networkUrl is directly used by the `findById` as `self-link`aka the networkUrl is the `@Id` to identify the `NetworkResource` if present.

This change will also mean that all the places where `NetworkResource` is used like https://github.com/perfectsense/gyro-google-provider/blob/feature/dns/src/main/java/gyro/google/compute/SubnetworkResource.java#L199 will now be using the `self-link` field as is used in this complex-type as well their respective `copyFrom()` method needs to set the networkUrl directly and not strip out the name and use that like https://github.com/perfectsense/gyro-google-provider/blob/feature/dns/src/main/java/gyro/google/compute/SubnetworkResource.java#L177-L179